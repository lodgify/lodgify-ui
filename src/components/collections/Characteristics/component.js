import React from 'react';
import PropTypes from 'prop-types';
import { List } from 'semantic-ui-react';

import { buildKeyFromStrings } from 'utils/build-key-from-strings';
import { getFirstFourItems } from 'utils/get-first-four-items';
import { Grid } from 'layout/Grid';
import { GridColumn } from 'layout/GridColumn';
import { Icon } from 'elements/Icon';
import { ShowOn } from 'layout/ShowOn';

/**
 * The standard widget for displaying a collection of characteristics for a property or a room.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({ spaceCharacteristics }) => (
  <Grid columns={1}>
    <GridColumn width={12}>
      <ShowOn
        computer
        parent={List}
        parentProps={{ horizontal: true }}
        tablet
        widescreen
      >
        {getFirstFourItems(spaceCharacteristics).map(
          ({ iconName, text }, index) => (
            <List.Item key={buildKeyFromStrings(text, index)}>
              <Icon labelText={text} name={iconName} />
            </List.Item>
          )
        )}
      </ShowOn>
      <ShowOn mobile parent={Grid} parentProps={{ columns: 1 }}>
        {getFirstFourItems(spaceCharacteristics).map(
          ({ iconName, text }, index) => (
            <GridColumn
              computer={3}
              key={buildKeyFromStrings(text, index)}
              mobile={6}
            >
              <Icon labelText={text} name={iconName} />
            </GridColumn>
          )
        )}
      </ShowOn>
    </GridColumn>
  </Grid>
);

Component.displayName = 'Characteristics';

Component.defaultProps = {
  spaceCharacteristics: [],
};

Component.propTypes = {
  /** The main characteristics to display. Maximum four. */
  spaceCharacteristics: PropTypes.arrayOf(
    PropTypes.shape({
      /**
       * The name of the icon to display.
       * [See here for the full list of valid icon names](https://github.com/lodgify/lodgify-ui/blob/master/src/components/elements/Icon/constants.js)
       */
      iconName: PropTypes.string.isRequired,
      /** A visible label to display for the key fact. */
      text: PropTypes.string.isRequired,
    })
  ),
};
