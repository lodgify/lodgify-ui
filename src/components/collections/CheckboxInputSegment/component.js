import React from 'react';
import PropTypes from 'prop-types';
import { Segment } from 'semantic-ui-react';

import { Divider } from 'elements/Divider';
import { FlexContainer } from 'layout/FlexContainer';
import { Grid } from 'layout/Grid';
import { Heading } from 'typography/Heading';
import { Checkbox } from 'inputs/Checkbox';
import { GridColumn } from 'layout/GridColumn';
import { withResponsive } from 'utils/with-responsive';
import { buildKeyFromStrings } from 'utils/build-key-from-strings';

import { getGridColumns } from './utils/getGridColumns';

/**
 * A collection of information alongside checkboxes.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
const Component = ({ heading, checkboxes, isUserOnMobile }) => (
  <Segment className="is-input-segment is-checkbox-input-segment" vertical>
    <FlexContainer
      alignItems="baseline"
      flexDirection="column"
      justifyContent="space-between"
    >
      <Heading>{heading}</Heading>
      <Divider size="small" />
      <Grid columns={getGridColumns(isUserOnMobile)}>
        {checkboxes.map(({ name, label, ...checkboxProps }) => (
          <GridColumn key={buildKeyFromStrings(label, name)}>
            <Checkbox
              label={label}
              name={name}
              {...checkboxProps}
              isFluid={isUserOnMobile}
              isLabelLeft={isUserOnMobile}
            />
          </GridColumn>
        ))}
      </Grid>
    </FlexContainer>
  </Segment>
);

Component.displayName = 'CheckboxInputSegment';

Component.propTypes = {
  /** The checkboxes to display. See [the `Checkbox` component for valid props](#/Inputs/Checkbox).  */
  checkboxes: PropTypes.arrayOf(PropTypes.object).isRequired,
  /** The text for the input segment's heading. */
  heading: PropTypes.string.isRequired,
  /**
   * Is the user on a mobile device.
   * Provided by `withResponsive` so ignored in the styleguide.
   * @ignore
   */
  isUserOnMobile: PropTypes.bool.isRequired,
};

export const ComponentWithResponsive = withResponsive(Component);
