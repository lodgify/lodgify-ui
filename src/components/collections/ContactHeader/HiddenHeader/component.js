import React, { useRef, useState, useEffect } from 'react';
import {
  string,
  arrayOf,
  shape,
  func,
  bool,
  oneOfType,
  number,
} from 'prop-types';
import { Menu, Accordion } from 'semantic-ui-react';
import { size } from 'lodash';

import { Heading } from 'typography/Heading';
import { buildKeyFromStrings } from 'utils/build-key-from-strings';
import { Icon, ICON_NAMES } from 'elements/Icon';
import { Modal } from 'elements/Modal';
import { Submenu } from 'elements/Submenu';
import { testidFactory } from 'utils/testid';

import { PHONE_PREFIX } from '../constants';

const TEST_ID_PREFIX = 'hiddenHeader';

const testid = testidFactory(TEST_ID_PREFIX);

export const Component = ({
  headingHref,
  headingText,
  navigationItems,
  languageOptions,
  onChangeLanguage,
  languageValue,
  currencyOptions,
  currencyNoResultsText,
  onChangeCurrency,
  currencyValue,
  phoneNumber,
}) => {
  const [activeItem, setActiveItem] = useState('');

  useEffect(() => {
    /* istanbul ignore next */
    if (navigationItems.length > 0) {
      setActiveItem(navigationItems[0].text);
    }
  }, [navigationItems.length]);

  const menu = useRef();

  const onMenuLinkClick = text => {
    setActiveItem(text);
  };

  return (
    <Modal
      className="hidden-contact-header-modal"
      closeIcon={<Icon labelText="Close" name="close" />}
      data-testid={testid()}
      header={
        <Menu.Item href={headingHref} link>
          <Heading className="header-title" size="small">
            {headingText}
          </Heading>
        </Menu.Item>
      }
      isFullscreen
      trigger={<Icon name={ICON_NAMES.BARS} />}
    >
      <div ref={menu}>
        <Menu text vertical>
          {navigationItems.map(({ subItems, text, link }, index) =>
            size(subItems) ? (
              <Accordion
                as={Menu.Item}
                key={buildKeyFromStrings(text, index)}
                panels={[
                  {
                    title: {
                      content: text,
                      key: buildKeyFromStrings(text, index),
                    },
                    content: {
                      content: subItems.map(({ text, href }, index) => (
                        <Menu.Item
                          active={text === activeItem}
                          data-testid={testid(`accordionItem_${index}`)}
                          href={href}
                          key={buildKeyFromStrings(text, index)}
                          link
                          onClick={() => onMenuLinkClick(text)}
                        >
                          {text}
                        </Menu.Item>
                      )),
                      key: buildKeyFromStrings(index, text),
                    },
                  },
                ]}
              />
            ) : (
              <Menu.Item
                active={text === activeItem}
                data-testid={testid(`hiddenHeaderItem_${index}`)}
                href={link}
                key={buildKeyFromStrings(text, index)}
                link
                onClick={() => onMenuLinkClick(text)}
              >
                {text}
              </Menu.Item>
            )
          )}
        </Menu>
        <Menu vertical>
          {size(languageOptions) > 1 && (
            <Menu.Item>
              <Submenu
                items={languageOptions}
                name="language"
                onChange={onChangeLanguage}
                value={languageValue}
                willOpenAbove
              />
            </Menu.Item>
          )}
          {size(currencyOptions) > 1 && (
            <Menu.Item>
              <Submenu
                isSearchable
                items={currencyOptions}
                name="currency"
                noResultsText={currencyNoResultsText}
                onChange={onChangeCurrency}
                value={currencyValue}
                willOpenAbove
              />
            </Menu.Item>
          )}
          {!!phoneNumber && (
            <Menu.Item className="is-selectable">
              <a href={`${PHONE_PREFIX}${phoneNumber.replace(/\s/g, '')}`}>
                <Icon labelText={phoneNumber} name={ICON_NAMES.PHONE} />
              </a>
            </Menu.Item>
          )}
        </Menu>
      </div>
    </Modal>
  );
};

Component.defaultProps = {
  currencyNoResultsText: '',
  currencyOptions: [],
  currencyValue: null,
  headingHref: '',
  headingText: '',
  languageOptions: [],
  languageValue: null,
  phoneNumber: '',
  onChangeCurrency: undefined,
  onChangeLanguage: undefined,
};

Component.propTypes = {
  /** The text to display when no results are returned from the currency dropdown. */
  currencyNoResultsText: string,
  /** The options which the user can select for the currency dropdown. */
  currencyOptions: arrayOf(
    shape({
      /** The visible text for the option. */
      text: string.isRequired,
      /** The underlying value for the option. */
      value: oneOfType([bool, number, string]),
    })
  ),
  /** The current value of the currency dropdown. */
  currencyValue: oneOfType([bool, number, string]),
  /** The link for the heading. */
  headingHref: string,
  /** The text for the heading. */
  headingText: string,
  /** The options which the user can select for the language dropdown. */
  languageOptions: arrayOf(
    shape({
      /** The visible text for the option. */
      text: string.isRequired,
      /** The underlying value for the option. */
      value: oneOfType([bool, number, string]),
    })
  ),
  /** The current value of the language dropdown. */
  languageValue: oneOfType([bool, number, string]),
  /** The items for a user to navigate the site. */
  navigationItems: arrayOf(
    shape({
      /** The href url for an item which is a link. */
      href: string,
      /** Sub navigation items to display as a [`<Submenu />`](#submenu) under an item. */
      subItems: arrayOf(
        shape({
          href: string.isRequired,
          target: string,
          text: string.isRequired,
        })
      ),
      /** Specifies where to display the linked navigation items URL. See [MDN docs `<a />` for more](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-target). */
      target: string,
      /** The visible text for an item. */
      text: string.isRequired,
    })
  ).isRequired,
  /** The function called when the currency dropdown is changed.
   *  @param {String} name
   *  @param {String} value
   */
  onChangeCurrency: func,
  /** The function called when the language dropdown is changed.
   *  @param {String} name
   *  @param {String} value
   */
  onChangeLanguage: func,
  /** The phone number to display. */
  phoneNumber: string,
};
