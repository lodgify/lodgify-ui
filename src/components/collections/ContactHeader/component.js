import React from 'react';
import { Menu } from 'semantic-ui-react';
import {
  string,
  arrayOf,
  shape,
  oneOfType,
  oneOf,
  number,
  bool,
  func,
} from 'prop-types';

import { FlexContainer } from 'layout/FlexContainer';
import { Button } from 'elements/Button';
import { Submenu } from 'elements/Submenu';
import { size } from 'utils/size';
import { Icon, ICON_NAMES } from 'elements/Icon';
import { ShowOn } from 'layout/ShowOn';
import { testidFactory } from 'utils/testid';
import { Logo } from 'elements/Logo';

import { HorizontalMenu } from '../HorizontalMenu';

import { MobileHeader } from './MobileHeader';
import { PHONE_PREFIX } from './constants';

const TEST_ID_PREFIX = 'contactHeader';

const testid = testidFactory(TEST_ID_PREFIX);

/**
 * A header displays a logo, grouped navigation items
 * contact information and an a primary call to action.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  primaryButtonText,
  navigationMenuItems,
  languageOptions,
  onChangeLanguage,
  currencyOptions,
  onChangeCurrency,
  currencyValue,
  currencyNoResultsText,
  languageValue,
  phoneNumber,
  logoHref,
  logoSubText,
  logoSize,
  logoSrc,
  logoSizes,
  logoText,
  logoSrcSet,
}) => (
  <header className="contact-header" data-testid={testid()}>
    <ShowOn computer tablet widescreen>
      <FlexContainer alignItems="center" justifyContent="space-between">
        <Menu>
          {size(languageOptions) > 1 && (
            <Menu.Item>
              <Submenu
                items={languageOptions}
                name="language"
                onChange={onChangeLanguage}
                value={languageValue}
                willOpenAbove
              />
            </Menu.Item>
          )}
          {size(currencyOptions) > 1 && (
            <Menu.Item>
              <Submenu
                isSearchable
                items={currencyOptions}
                name="currency"
                noResultsText={currencyNoResultsText}
                onChange={onChangeCurrency}
                value={currencyValue}
                willOpenAbove
              />
            </Menu.Item>
          )}
          {!!phoneNumber && (
            <Menu.Item className="is-selectable">
              <a href={`${PHONE_PREFIX}${phoneNumber.replace(/\s/g, '')}`}>
                <Icon labelText={phoneNumber} name={ICON_NAMES.PHONE} />
              </a>
            </Menu.Item>
          )}
        </Menu>
        <Logo
          logoHref={logoHref}
          logoSize={logoSize}
          logoSizes={logoSizes}
          logoSrc={logoSrc}
          logoSrcSet={logoSrcSet}
          logoSubText={logoSubText}
          logoSubTextAlignment="center"
          logoText={logoText}
        />
        <Button>{primaryButtonText}</Button>
      </FlexContainer>
      <HorizontalMenu isHeader items={navigationMenuItems} />
    </ShowOn>
    <ShowOn mobile>
      <FlexContainer
        alignItems="center"
        className="hidden-contact-header"
        justifyContent="space-between"
      >
        <Logo
          logoHref={logoHref}
          logoSize={logoSize}
          logoSizes={logoSizes}
          logoSrc={logoSrc}
          logoSrcSet={logoSrcSet}
          logoSubText={logoSubText}
          logoText={logoText}
        />
        <MobileHeader
          currencyNoResultsText={currencyNoResultsText}
          currencyOptions={currencyOptions}
          currencyValue={currencyValue}
          languageOptions={languageOptions}
          languageValue={languageValue}
          logoHref={logoHref}
          logoSize={logoSize}
          logoSizes={logoSizes}
          logoSrc={logoSrc}
          logoSrcSet={logoSrcSet}
          logoSubText={logoSubText}
          logoText={logoText}
          navigationItems={navigationMenuItems}
          onChangeCurrency={onChangeCurrency}
          phoneNumber={phoneNumber}
        />
      </FlexContainer>
    </ShowOn>
  </header>
);

Component.displayName = 'ContactHeader';

Component.defaultProps = {
  currencyNoResultsText: undefined,
  currencyOptions: [],
  currencyValue: null,
  languageOptions: [],
  languageValue: null,
  logoHref: '',
  logoSize: 'medium',
  logoSizes: undefined,
  logoSrc: undefined,
  logoSrcSet: undefined,
  logoSubText: '',
  logoText: '',
  onChangeCurrency: undefined,
  onChangeLanguage: undefined,
  phoneNumber: '',
  primaryButtonText: '',
};

Component.propTypes = {
  /** The text to display when no results are returned from the currency dropdown. */
  currencyNoResultsText: string,
  /** The options which the user can select for the currency dropdown. */
  currencyOptions: arrayOf(
    shape({
      /** The visible text for the option. */
      text: string.isRequired,
      /** The underlying value for the option. */
      value: oneOfType([bool, number, string]),
    })
  ),
  /** The current value of the currency dropdown. */
  currencyValue: oneOfType([bool, number, string]),
  /** The options which the user can select for the language dropdown. */
  languageOptions: arrayOf(
    shape({
      /** The visible text for the option. */
      text: string.isRequired,
      /** The underlying value for the option. */
      value: oneOfType([bool, number, string]),
    })
  ),
  /** The current value of the language dropdown. */
  languageValue: oneOfType([bool, number, string]),
  /** The href for the logo link. */
  logoHref: string,
  /** The maximum size of the headers logo. */
  logoSize: oneOf(['medium', 'large', 'huge']),
  /** A list of one or more strings separated by commas indicating a set of source sizes for the logo. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
  logoSizes: string,
  /** The src url for the logo. */
  logoSrc: string,
  /** A list of one or more strings separated by commas indicating a set of possible image sources for the user agent to use for the logo. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
  logoSrcSet: string,
  /** The text that appears under the logo or logo text. */
  logoSubText: string,
  /** The text for the logo. */
  logoText: string,
  /** The items for a user to navigate the site. */
  navigationMenuItems: arrayOf(
    shape({
      /** The href url for an item which is a link. */
      href: string,
      /** Sub navigation items to display as a [`<Submenu />`](#submenu) under an item. */
      subItems: arrayOf(
        shape({
          href: string.isRequired,
          target: string,
          text: string.isRequired,
        })
      ),
      /** Specifies where to display the linked navigation items URL. See [MDN docs `<a />` for more](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-target). */
      target: string,
      /** The visible text for an item. */
      text: string.isRequired,
    })
  ).isRequired,
  /** The function called when the currency dropdown is changed.
   *  @param {String} name
   *  @param {String} value
   */
  onChangeCurrency: func,
  /** The function called when the language dropdown is changed.
   *  @param {String} name
   *  @param {String} value
   */
  onChangeLanguage: func,
  /** The phone number to display. */
  phoneNumber: string,
  /** The text to display for the primary button. */
  primaryButtonText: string,
};
