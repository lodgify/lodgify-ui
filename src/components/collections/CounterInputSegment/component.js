import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Segment } from 'semantic-ui-react';

import { Counter } from 'inputs/Counter';
import { Divider } from 'elements/Divider';
import { FlexContainer } from 'layout/FlexContainer';
import { Heading } from 'typography/Heading';
import { Paragraph } from 'typography/Paragraph';
import { buildKeyFromStrings } from 'utils/build-key-from-strings';

/**
 * A collection of information alongside a counter inputs.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({ heading, counters }) => (
  <Segment className="is-input-segment is-counter-input-segment" vertical>
    <FlexContainer
      alignItems="baseline"
      flexDirection="column"
      justifyContent="space-between"
    >
      <Heading>{heading}</Heading>
      <Divider size="small" />
      {counters.map(({ name, ...counterProps }, index) => (
        <Fragment key={buildKeyFromStrings(name, index)}>
          <FlexContainer flexDirection="row" justifyContent="space-between">
            <Paragraph>{name}</Paragraph>
            <Counter name={name} {...counterProps} />
          </FlexContainer>
          <Divider size="small" />
        </Fragment>
      ))}
    </FlexContainer>
  </Segment>
);

Component.displayName = 'CounterInputSegment';

Component.propTypes = {
  /**	An array of [`Counter`](#/Counter) props objects. */
  counters: PropTypes.arrayOf(PropTypes.object).isRequired,
  /** The text for the input segment's heading. */
  heading: PropTypes.string.isRequired,
};
