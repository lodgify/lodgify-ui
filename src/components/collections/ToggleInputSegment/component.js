import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Segment } from 'semantic-ui-react';

import { Toggle } from 'inputs/Toggle';
import { FlexContainer } from 'layout/FlexContainer';
import { Heading } from 'typography/Heading';
import { Paragraph } from 'typography/Paragraph';

/**
 * A collection of information alongside a toggle input.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export class Component extends PureComponent {
  state = {
    isToggleChecked: this.props.isToggleChecked || false,
  };

  handleOnClick = event => {
    const { isToggleChecked } = this.state;

    this.setState({
      isToggleChecked: this.props.isToggleChecked || !isToggleChecked,
    });
    this.props.onClick(event, isToggleChecked);
  };

  render = () => {
    const { heading, description } = this.props;

    return (
      <Segment
        className="is-input-segment is-toggle-input-segment"
        onClick={this.handleOnClick}
        vertical
      >
        <FlexContainer
          alignItems="baseline"
          flexDirection="row"
          justifyContent="space-between"
        >
          <div>
            <Heading>{heading}</Heading>
            {!!description && <Paragraph>{description}</Paragraph>}
          </div>
          <Toggle isChecked={this.state.isToggleChecked} isCompact />
        </FlexContainer>
      </Segment>
    );
  };
}

Component.displayName = 'ToggleInputSegment';

Component.defaultProps = {
  description: null,
  onClick: Function.prototype,
  isToggleChecked: false,
};

Component.propTypes = {
  /** The text for the input segment's description. */
  description: PropTypes.string,
  /** The text for the input segment's heading. */
  heading: PropTypes.string.isRequired,
  /** Is the toggle checked. */
  isToggleChecked: PropTypes.bool,
  /**
   * Event called when the input segment is clicked.
   * @param {Object} event
   * @param {bool}   isToggleChecked
   */
  onClick: PropTypes.func,
};
