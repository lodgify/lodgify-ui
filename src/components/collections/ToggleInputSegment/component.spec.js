import React from 'react';
import { mount, shallow } from 'enzyme';
import { expectComponentToHaveDisplayName } from '@lodgify/enzyme-jest-expect-helpers';

import { Component } from './component';

const heading = '👤';

const getToggleInputSegment = props =>
  mount(<Component heading={heading} {...props} />);

const getShallowToggleInputSegment = props =>
  shallow(<Component heading={heading} {...props} />);

describe('Component', () => {
  describe('by default', () => {
    it('should render the right structure', () => {
      const actual = getToggleInputSegment();

      expect(actual).toMatchSnapshot();
    });
  });

  describe('if `props.description` is passed', () => {
    it('should render the right structure', () => {
      const actual = getToggleInputSegment({ description: '🐷' });

      expect(actual).toMatchSnapshot();
    });
  });

  describe('componentDidUpdate', () => {
    describe('if a controlled `isToggleChecked` value has changed', () => {
      it('should call `setState` and `props.onChange` with the right arguments', () => {
        const name = 'some name';
        const isToggleChecked = true;
        const onChange = jest.fn();
        const setState = jest.fn();
        const props = {
          name,
          onChange,
          isToggleChecked,
        };
        const wrapper = getShallowToggleInputSegment(props);

        wrapper.instance().setState = setState;
        wrapper.instance().componentDidUpdate({ isToggleChecked: false }, {});

        expect(setState).toHaveBeenCalledWith({
          isToggleChecked: props.isToggleChecked,
        });
        expect(onChange).toHaveBeenCalledWith(name, isToggleChecked);
      });
    });

    describe('if a uncontrolled `isToggleChecked` value has changed', () => {
      it('should call `onChange` with the right arguments', () => {
        const name = 'some name';
        const onChange = jest.fn();
        const props = {
          name,
          onChange,
        };
        const wrapper = getShallowToggleInputSegment(props);

        wrapper
          .instance()
          .componentDidUpdate(
            { isToggleChecked: false },
            { isToggleChecked: true }
          );

        expect(onChange).toHaveBeenCalledWith(
          name,
          wrapper.instance().state.isToggleChecked
        );
      });
    });

    describe('if neither `this.props.isToggleChecked` or `this.state.isToggleChecked` have changed', () => {
      it('should not call `props.onChange` or `setState`', () => {
        const onChange = jest.fn();
        const setState = jest.fn();

        const wrapper = getShallowToggleInputSegment({
          onChange,
        });

        wrapper.instance().setState = setState;
        wrapper
          .instance()
          .componentDidUpdate(
            { isToggleChecked: false },
            { isToggleChecked: false }
          );

        expect(onChange).not.toHaveBeenCalled();
        expect(setState).not.toHaveBeenCalled();
      });
    });
  });

  describe('handleOnClick', () => {
    it('should called `setState` with the correct arguments', () => {
      const wrapper = getToggleInputSegment();

      wrapper.instance().setState = jest.fn();
      wrapper.instance().handleOnClick();

      expect(wrapper.instance().setState).toHaveBeenCalledWith({
        isToggleChecked: !wrapper.instance().state.isToggleChecked,
      });
    });

    describe('if `this.props.isToggleChecked` === true', () => {
      it('should called `setState` with the correct arguments', () => {
        const wrapper = getToggleInputSegment({ isToggleChecked: true });

        wrapper.instance().setState = jest.fn();
        wrapper.instance().handleOnClick();

        expect(wrapper.instance().setState).toHaveBeenCalledWith({
          isToggleChecked: wrapper.instance().props.isToggleChecked,
        });
      });
    });

    it('should call `props.onClick` with the correct arguments', () => {
      const event = {
        '🎇': '🎆',
      };

      const wrapper = getToggleInputSegment({
        onClick: jest.fn(),
        isToggleChecked: true,
      });

      wrapper.instance().handleOnClick(event);

      expect(wrapper.instance().props.onClick).toHaveBeenCalledWith(event);
    });
  });

  it('should have the correct `displayName`', () => {
    expectComponentToHaveDisplayName(Component, 'ToggleInputSegment');
  });
});
