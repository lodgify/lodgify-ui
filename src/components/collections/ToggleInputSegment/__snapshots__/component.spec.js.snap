// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component by default should render the right structure 1`] = `
<ToggleInputSegment
  description={null}
  heading="üë§"
  isToggleChecked={false}
  onClick={[Function]}
>
  <Segment
    className="is-input-segment is-toggle-input-segment"
    onClick={[Function]}
    vertical={true}
  >
    <div
      className="ui vertical segment is-input-segment is-toggle-input-segment"
      onClick={[Function]}
    >
      <FlexContainer
        alignContent={null}
        alignItems="baseline"
        flexDirection="row"
        flexWrap={null}
        justifyContent="space-between"
      >
        <div
          className="flex-container"
          style={
            Object {
              "alignContent": null,
              "alignItems": "baseline",
              "display": "flex",
              "flexDirection": "row",
              "flexGrow": "1",
              "flexWrap": null,
              "height": "100%",
              "justifyContent": "space-between",
            }
          }
        >
          <div>
            <Heading
              className={null}
              hasMargin={true}
              isColorInverted={false}
              size="medium"
              textAlign={null}
            >
              <Header
                as="h3"
                className=""
                inverted={false}
                textAlign={null}
              >
                <h3
                  className="ui header"
                >
                  üë§
                </h3>
              </Header>
            </Heading>
          </div>
          <Toggle
            isChecked={false}
            isCompact={true}
          >
            <Checkbox
              error={false}
              isChecked={false}
              isCompact={true}
              isDisabled={false}
              isFluid={false}
              isLabelLeft={false}
              isRadioButton={false}
              isToggle={true}
              label=""
              name=""
              onChange={[Function]}
              onClick={[Function]}
            >
              <InputController
                adaptOnChangeEvent={[Function]}
                error={false}
                icon={null}
                initialValue=""
                inputOnChangeFunctionName="onChange"
                isCompact={true}
                isFluid={false}
                isFocused={false}
                isValid={false}
                label={null}
                mapValueToProps={[Function]}
                name=""
                onChange={[Function]}
                value={false}
              >
                <Input
                  className="compact"
                  fluid={false}
                  iconPosition={null}
                  type="text"
                >
                  <div
                    className="ui input compact"
                  >
                    <Checkbox
                      checked={false}
                      className=""
                      disabled={false}
                      key=".0"
                      label=""
                      onChange={[Function]}
                      onClick={[Function]}
                      radio={false}
                      toggle={true}
                      type="checkbox"
                    >
                      <div
                        className="ui toggle checkbox"
                        onChange={[Function]}
                        onClick={[Function]}
                        onMouseDown={[Function]}
                      >
                        <input
                          checked={false}
                          className="hidden"
                          onClick={[Function]}
                          readOnly={true}
                          tabIndex={0}
                          type="checkbox"
                        />
                        <label
                          key=""
                        />
                      </div>
                    </Checkbox>
                  </div>
                </Input>
              </InputController>
            </Checkbox>
          </Toggle>
        </div>
      </FlexContainer>
    </div>
  </Segment>
</ToggleInputSegment>
`;

exports[`Component if \`props.description\` is passed should render the right structure 1`] = `
<ToggleInputSegment
  description="üê∑"
  heading="üë§"
  isToggleChecked={false}
  onClick={[Function]}
>
  <Segment
    className="is-input-segment is-toggle-input-segment"
    onClick={[Function]}
    vertical={true}
  >
    <div
      className="ui vertical segment is-input-segment is-toggle-input-segment"
      onClick={[Function]}
    >
      <FlexContainer
        alignContent={null}
        alignItems="baseline"
        flexDirection="row"
        flexWrap={null}
        justifyContent="space-between"
      >
        <div
          className="flex-container"
          style={
            Object {
              "alignContent": null,
              "alignItems": "baseline",
              "display": "flex",
              "flexDirection": "row",
              "flexGrow": "1",
              "flexWrap": null,
              "height": "100%",
              "justifyContent": "space-between",
            }
          }
        >
          <div>
            <Heading
              className={null}
              hasMargin={true}
              isColorInverted={false}
              size="medium"
              textAlign={null}
            >
              <Header
                as="h3"
                className=""
                inverted={false}
                textAlign={null}
              >
                <h3
                  className="ui header"
                >
                  üë§
                </h3>
              </Header>
            </Heading>
            <Paragraph
              size="medium"
              weight={null}
            >
              <p
                className=""
              >
                üê∑
              </p>
            </Paragraph>
          </div>
          <Toggle
            isChecked={false}
            isCompact={true}
          >
            <Checkbox
              error={false}
              isChecked={false}
              isCompact={true}
              isDisabled={false}
              isFluid={false}
              isLabelLeft={false}
              isRadioButton={false}
              isToggle={true}
              label=""
              name=""
              onChange={[Function]}
              onClick={[Function]}
            >
              <InputController
                adaptOnChangeEvent={[Function]}
                error={false}
                icon={null}
                initialValue=""
                inputOnChangeFunctionName="onChange"
                isCompact={true}
                isFluid={false}
                isFocused={false}
                isValid={false}
                label={null}
                mapValueToProps={[Function]}
                name=""
                onChange={[Function]}
                value={false}
              >
                <Input
                  className="compact"
                  fluid={false}
                  iconPosition={null}
                  type="text"
                >
                  <div
                    className="ui input compact"
                  >
                    <Checkbox
                      checked={false}
                      className=""
                      disabled={false}
                      key=".0"
                      label=""
                      onChange={[Function]}
                      onClick={[Function]}
                      radio={false}
                      toggle={true}
                      type="checkbox"
                    >
                      <div
                        className="ui toggle checkbox"
                        onChange={[Function]}
                        onClick={[Function]}
                        onMouseDown={[Function]}
                      >
                        <input
                          checked={false}
                          className="hidden"
                          onClick={[Function]}
                          readOnly={true}
                          tabIndex={0}
                          type="checkbox"
                        />
                        <label
                          key=""
                        />
                      </div>
                    </Checkbox>
                  </div>
                </Input>
              </InputController>
            </Checkbox>
          </Toggle>
        </div>
      </FlexContainer>
    </div>
  </Segment>
</ToggleInputSegment>
`;
